devfiles/nodejs/devfile.yaml

```diff
@@ -3,34 +3,52 @@ metadata:
   name: nodejs
   version: 1.0.0
 projects:
-  - name: nodejs-web-app
+  - name: nodejs-starter
     git:
       location: "https://github.com/odo-devfiles/nodejs-ex.git"
 components:
   - container:
       name: runtime
-      image: quay.io/eclipse/che-nodejs10-ubi:nightly
+      image: registry.access.redhat.com/ubi8/nodejs-12:1-36
       memoryLimit: 1024Mi
       mountSources: true
       endpoints:
-        - name: "http-3000"
-          configuration:
-            public: true
+        - name: http-3000
           targetPort: 3000
+          configuration:
+            protocol: tcp
+            scheme: http
+            type: terminal
 commands:
   - exec:
-      id: install
+      id: download app dependencies
```

Comment:
Consider to have shorter command id, this is in preparation for the future that odo may provide a way for user to execute any command on the devfile. When we have that, the user will be using the id to indicate which command to run on. A shorter name (preferably without space) will help user to call those command easily in the future.